src/app/components/EditorInterface.tsx

```diff
@@ -161,19 +161,27 @@ const EditorInterface = ({ settings, token }: IEditorInterface) => {
       )
       //add comma separator for custom fields
       const customFields: any = _client?.customFields
+
+      // Iterate through each key in customFields
       for (const key in customFields) {
-        if (Array.isArray(customFields[key])) {
-          //element[0].toUpperCase() + element.substring(1) is a hack to capitalize the first string, however changes in SDK response
-          //is required.
-          customFields[key] = customFields[key].map(
-            (element: any) =>
-              ' ' + element[0].toUpperCase() + element.substring(1),
-          )
+        // Check if the value is an array and if the key exists in allCustomFields
+        if (
+          Array.isArray(customFields[key]) &&
+          appState?.appState.customFields.some((field) => field.key === key)
+        ) {
+          // Map the values to their corresponding labels
+          customFields[key] = customFields[key].map((value: any) => {
+            const option: any = (appState?.appState?.customFields as any)
```

Comment:
can be define types instead of any

Reply:
add types to `value` in the next commit. `any` type is assigned to the customField because customField can have any structure. This can be improved but need updates in different parts of the code which I will do while doing [THIS](https://linear.app/copilotplatforms/issue/OUT-72/miscellaneous-client-home-cleanup) task. @sazanrjb

Reply:
if we dont know the type, then lets use `unknown`. Lets do this is another PR then.